mongo/jobs.go

```diff
@@ -241,3 +197,30 @@ func (m *JobStore) UpdateJob(ctx context.Context, id string, updates bson.M) err
 
 	return nil
 }
+
+// ValidateJobIDUnique checks if there exists a reindex job in mongo with the given id
+func (m *JobStore) ValidateJobIDUnique(ctx context.Context, id string) error {
```

Comment:
Why was this created? The generation of ids for documents will be unique we chose the function so it's on us; by adding this in we are adding latency to the system for no added value

Reply:
Also we are not locking the collection down, so who is to say that once this returns a request to the database does not come in to create a document with the same unique id this validation process is using?

I guess the lock will actually exist in the calling method but still i dont see it.

Anyway to make life simpler i would say this method is not needed, mongo will complain when creating a resource if the id is not unique

Reply:
Done